### PR TITLE
KOGITO-1157: [DMN Designer] New models must be named with the file name on VSCode

### PR DESCRIPTION
**Pull request by danielzhe**
See: [https://issues.redhat.com/browse/KOGITO-1157](https://issues.redhat.com/browse/KOGITO-1157)

### CODE REVIEW

File: kie-wb-common-dmn/kie-wb-common-dmn-webapp-kogito-common/src/test/java/org/kie/workbench/common/dmn/webapp/kogito/common/client/services/DMNClientDiagramServiceImplTest.java [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+    }
+
+    @Test
+    public void testTransformWhenFileIsNew() {
+        final String fileName = "file.dmn";
+        final String xml = "";
+        final ServiceCallback<Diagram> callback = mock(ServiceCallback.class);
+        final String title = "title";
+
+        doReturn(title).when(service).getDiagramTitle(fileName);
```
Comment (manstis) [inline kie-wb-common-dmn/kie-wb-common-dmn-webapp-kogito-common/src/test/java/org/kie/workbench/common/dmn/webapp/kogito/common/client/services/DMNClientDiagramServiceImplTest.java]: Do we really have to mock the behaviour of this method!?!

File: kie-wb-common-dmn/kie-wb-common-dmn-webapp-kogito-common/src/test/java/org/kie/workbench/common/dmn/webapp/kogito/common/client/services/DMNClientDiagramServiceImplTest.java [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+    }
+
+    @Test
+    public void testTransformWhenFileIsNotNew() {
+        final String fileName = "file.dmn";
+        final String xml = "xml";
+        final ServiceCallback<Diagram> callback = mock(ServiceCallback.class);
+        final String title = "title";
+
+        doReturn(title).when(service).getDiagramTitle(fileName);
```
Comment (manstis) [inline kie-wb-common-dmn/kie-wb-common-dmn-webapp-kogito-common/src/test/java/org/kie/workbench/common/dmn/webapp/kogito/common/client/services/DMNClientDiagramServiceImplTest.java]: Do we really have to mock the behaviour of this method!?!

File: kie-wb-common-dmn/kie-wb-common-dmn-webapp-kogito-common/src/main/java/org/kie/workbench/common/dmn/webapp/kogito/common/client/services/DMNClientDiagramServiceImpl.java [hunk 3]
```diff
@@ -95,17 +95,39 @@ public DMNClientDiagramServiceImpl(final DMNMarshallerKogitoUnmarshaller dmnMars
...
+        return UUID.uuid();
+    }
+
+    @Override
+    public void transform(final String xml,
+                          final ServiceCallback<Diagram> callback) {
+        transform(UUID.uuid(), xml, callback);
+    }
+
+    void doNewDiagram(final String title, final ServiceCallback<Diagram> callback) {
```
Comment (manstis) [inline kie-wb-common-dmn/kie-wb-common-dmn-webapp-kogito-common/src/main/java/org/kie/workbench/common/dmn/webapp/kogito/common/client/services/DMNClientDiagramServiceImpl.java]: I'm a stickler for parameters on new lines.. but it's just my opinion, so I state it but don't expect it to be fixed.

File: kie-wb-common-dmn/kie-wb-common-dmn-webapp-kogito-common/src/main/java/org/kie/workbench/common/dmn/webapp/kogito/common/client/services/DMNClientDiagramServiceImpl.java [hunk 4]
```diff
@@ -95,17 +95,39 @@ public DMNClientDiagramServiceImpl(final DMNMarshallerKogitoUnmarshaller dmnMars
     //Kogito requirements
 
     @Override
-    public void transform(final String xml,
-                          final ServiceCallback<Diagram> callback) {
+    public void transform(final String fileName, final String xml, final ServiceCallback<Diagram> callback) {
```
Comment (manstis) [inline kie-wb-common-dmn/kie-wb-common-dmn-webapp-kogito-common/src/main/java/org/kie/workbench/common/dmn/webapp/kogito/common/client/services/DMNClientDiagramServiceImpl.java]: I'm a stickler for parameters on new lines.. but it's just my opinion, so I state it but don't expect it to be fixed.

File: kie-wb-common-dmn/kie-wb-common-dmn-webapp-kogito-common/src/main/java/org/kie/workbench/common/dmn/webapp/kogito/common/client/services/DMNClientDiagramServiceImpl.java (hunk 3 above)
Comment (danielzhe) [reply kie-wb-common-dmn/kie-wb-common-dmn-webapp-kogito-common/src/main/java/org/kie/workbench/common/dmn/webapp/kogito/common/client/services/DMNClientDiagramServiceImpl.java]: Like you I also prefer parameters on new lines, but sometimes the reviewers don't, so it's hard to know what to do. I'm changing anyway.

**Comment by manstis** [comment]
Thanks @danielzhe.. I've already approved but appreciate the changes.

**Comment by jomarko** [comment]
I am nto able to verify this on VS Code. I am getting this when creating new file, any advice?
Screenshot from 2020-03-02 10-56-58

**Comment by jomarko** [comment]
@manstis ok, thank you I downloaded latest **kogito-tooling** and I am able to reproduce the same as  you in https://github.com/kiegroup/kie-wb-common/pull/3199#pullrequestreview-367084379

**Comment by danielzhe** [comment]
@manstis @jomarko Thank you. Change applied! In Windows the file always comes *without* the full path (only filename), but I changed now to handle the situation for Unix and if there is other unknown situation where file comes with full path in Windows too.

**Comment by danielzhe** [comment]
@manstis Tests merged. PR ready to be merged.

**Pull request merged**